detector/calorimeter/HCalThreePartsEndcap_o1_v01_geo.cpp

```diff
@@ -1,467 +1,496 @@
+static dd4hep::Ref_t createHCalEC(dd4hep::Detector& lcdd, xml_h xmlElement, dd4hep::SensitiveDetector sensDet){
+    xml_det_t xmlDet = xmlElement;
+    std::string detName = xmlDet.nameStr();
+
+    // Make DetElement
+    dd4hep::DetElement caloDetElem(detName, xmlDet.id());
+
+    // Make volume that envelopes the whole barrel; set material to air
+    Dimension dimensions(xmlDet.dimensions());
+
+    dd4hep::Tube envelope(dimensions.rmin(), dimensions.rmax1(), (dimensions.v_offset() + dimensions.z_length()));
+    dd4hep::Tube negative1(dimensions.rmin(), dimensions.rmax1(), (dimensions.offset() - dimensions.width()));
+    dd4hep::Tube negative2(dimensions.rmin(), dimensions.rmin1(), (dimensions.z_offset() - dimensions.dz()));
+    dd4hep::Tube negative3(dimensions.rmin(), dimensions.rmin2(), (dimensions.v_offset() - dimensions.z_length()));
+
+    dd4hep::SubtractionSolid envelopeShapeTmp1(envelope, negative1);
+    dd4hep::SubtractionSolid envelopeShapeTmp2(envelopeShapeTmp1, negative2);
+    dd4hep::SubtractionSolid envelopeShape(envelopeShapeTmp2, negative3);
+
+    Volume envelopeVolume(detName + "_volume", envelopeShape, lcdd.air());
+    envelopeVolume.setVisAttributes(lcdd, dimensions.visStr());
+
+    // Set sensitive detector type
+    Dimension sensDetType = xmlElement.child(_Unicode(sensitive));
+    sensDet.setType(sensDetType.typeStr());
+
+    // Dimension dimensions(xmlElement.child(_Unicode(dimensions)));
+    xml_comp_t xEndPlate = xmlElement.child(_Unicode(end_plate));
+    double dZEndPlate = xEndPlate.thickness() / 2.;
+    xml_comp_t xFacePlate = xmlElement.child(_Unicode(face_plate));
+    double dRhoFacePlate = xFacePlate.thickness() / 2.;
+    xml_comp_t xSpace = xmlElement.child(_Unicode(plate_space));  // to avoid overlaps
+    double space = xSpace.thickness();
+    xml_comp_t xSteelSupport = xmlElement.child(_Unicode(steel_support));
+    double dSteelSupport = xSteelSupport.thickness();
+    lLog << MSG::DEBUG << "steel support thickness " << dSteelSupport << endmsg;
+    lLog << MSG::DEBUG << "steel support material  " << xSteelSupport.materialStr() << endmsg;
+
+    // Calculate sensitive barrel dimensions
+    double sensitiveBarrel1Rmin = dimensions.rmin1() + 2 * dRhoFacePlate + space;
+    double sensitiveBarrel2Rmin = dimensions.rmin2() + 2 * dRhoFacePlate + space;
+    double sensitiveBarrel3Rmin = dimensions.rmin() + 2 * dRhoFacePlate + space;
+
+    // Offset in z is given as distance from 0 to the middle of the Calorimeter volume
+    double extBarrelOffset1 = dimensions.offset();
+    double extBarrelOffset2 = dimensions.z_offset();
+    double extBarrelOffset3 = dimensions.v_offset();
+
+    // Hard-coded assumption that we have two different sequences for the modules
+    std::vector<xml_comp_t> sequences = {xmlElement.child(_Unicode(sequence_a)), xmlElement.child(_Unicode(sequence_b))};
+    // NOTE: This assumes that both have the same dimensions!
```

Comment:
is it possible to make the code fail if sequences[0] and [1] have different dimensions?